nixos/tests/openssh.nix

```diff
@@ -58,47 +58,63 @@ in {
   };
 
   testScript = ''
-    startAll;
-
-    my $key=`${pkgs.openssh}/bin/ssh-keygen -t ed25519 -f key -N ""`;
-
-    $server->waitForUnit("sshd");
-
-    subtest "manual-authkey", sub {
-      $server->succeed("mkdir -m 700 /root/.ssh");
-      $server->copyFileFromHost("key.pub", "/root/.ssh/authorized_keys");
-      $server_lazy->succeed("mkdir -m 700 /root/.ssh");
-      $server_lazy->copyFileFromHost("key.pub", "/root/.ssh/authorized_keys");
-
-      $client->succeed("mkdir -m 700 /root/.ssh");
-      $client->copyFileFromHost("key", "/root/.ssh/id_ed25519");
-      $client->succeed("chmod 600 /root/.ssh/id_ed25519");
-
-      $client->waitForUnit("network.target");
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no server 'echo hello world' >&2");
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no server 'ulimit -l' | grep 1024");
-
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no server_lazy 'echo hello world' >&2");
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no server_lazy 'ulimit -l' | grep 1024");
-
-    };
-
-    subtest "configured-authkey", sub {
-      $client->succeed("cat ${snakeOilPrivateKey} > privkey.snakeoil");
-      $client->succeed("chmod 600 privkey.snakeoil");
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null" .
-                       " -o StrictHostKeyChecking=no -i privkey.snakeoil" .
-                       " server true");
-
-      $client->succeed("ssh -o UserKnownHostsFile=/dev/null" .
-                       " -o StrictHostKeyChecking=no -i privkey.snakeoil" .
-                       " server_lazy true");
-
-    };
-
-    subtest "localhost-only", sub {
-      $server_localhost_only->succeed("ss -nlt | grep '127.0.0.1:22'");
-      $server_localhost_only_lazy->succeed("ss -nlt | grep '127.0.0.1:22'");
-    }
+    from subprocess import call
+
+    start_all()
+
+    call(
+        [
+            "${pkgs.openssh}/bin/ssh-keygen",
+            "-t",
+            "ed25519",
+            "-f",
+            "key",
+            "-N",
+            "",
+        ]
+    )
+
+    server.wait_for_unit("sshd")
+
+    with subtest("manual-authkey"):
+        server.succeed("mkdir -m 700 /root/.ssh")
+        server.copy_file_from_host("key.pub", "/root/.ssh/authorized_keys")
+        server_lazy.succeed("mkdir -m 700 /root/.ssh")
+        server_lazy.copy_file_from_host("key.pub", "/root/.ssh/authorized_keys")
+
+        client.succeed("mkdir -m 700 /root/.ssh")
+        client.copy_file_from_host("key", "/root/.ssh/id_ed25519")
```

Comment:
`client.succeed` should return the command output. Could you try generating the key on the client, read the pubkey in the test script, and copy it to the server that way?

Reply:
Yes, I've pushed the changes.